adapters/repos/db/index_queue_test.go

```diff
@@ -0,0 +1,723 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/weaviate/weaviate/adapters/repos/db/helpers"
+	"github.com/weaviate/weaviate/adapters/repos/db/indexcheckpoint"
+	"github.com/weaviate/weaviate/adapters/repos/db/vector/hnsw/distancer"
+	"github.com/weaviate/weaviate/entities/storagestate"
+)
+
+func startWorker(t testing.TB, retryInterval ...time.Duration) chan job {
+	t.Helper()
+	ch := make(chan job)
+	t.Cleanup(func() {
+		close(ch)
+	})
+
+	itv := time.Millisecond
+	if len(retryInterval) > 0 {
+		itv = retryInterval[0]
+	}
+
+	go func() {
+		logger := logrus.New()
+		logger.Level = logrus.ErrorLevel
+		asyncWorker(ch, logger, itv)
+	}()
+
+	return ch
+}
+
+func newCheckpointManager(t testing.TB) *indexcheckpoint.Checkpoints {
+	t.Helper()
+
+	return newCheckpointManagerWithDir(t, t.TempDir())
+}
+
+func newCheckpointManagerWithDir(t testing.TB, dir string) *indexcheckpoint.Checkpoints {
+	t.Helper()
+
+	c, err := indexcheckpoint.New(dir, logrus.New())
+	require.NoError(t, err)
+
+	return c
+}
+
+func pushVector(t testing.TB, ctx context.Context, q *IndexQueue, id uint64, vector ...float32) {
```

Comment:
could you change  type of `vector` to `[]float32`? now it kinda blends with `id` when used (`pushVector(t, ctx, q, 1, 1, 2, 3)`)

to have `pushVector(t, ctx, q, 1, []float32{1, 2, 3})`,
or `pushVector(t, ctx, q, 1, vec(1, 2, 3))` (`vec` function needs to be created in that case)